static/js/publisher/release/actions/releases.js

```diff
@@ -1,5 +1,177 @@
+      json.closed_channels.forEach(channel => {
+        // make sure channels without track name get prefixed with 'latest'
+        if (RISKS.indexOf(channel.split("/")[0]) !== -1) {
+          // TODO: This should be the default track, not latest
+          channel = `latest/${channel}`;
+        }
+
+        closeChannelSuccess(channel);
+      });
+    }
+  } else {
+    let error = new Error(
+      `Error while closing channels: ${channels.join(", ")}.`
+    );
+    error.json = json;
+    throw error;
+  }
+}
+
+function fetchCloses(csrfToken, snapName, channels) {
+  if (channels.length) {
+    return fetchClose(channels).then(json => {
+      handleCloseResponse(json, channels);
+    });
+  } else {
+    return Promise.resolve();
+  }
+}
+
+function handleReleaseError(error) {
+  let message = error.message || ERROR_MESSAGE;
+
+  // try to find error messages in response json
+  // which may be an array or errors or object with errors propery
+  if (error.json) {
+    const errors = error.json.length ? error.json : error.json.errors;
+
+    if (errors.length) {
+      message = `${message} ${errors
+        .map(e => e.message)
+        .filter(m => m)
+        .join(" ")}`;
+    }
+  }
+
+  showNotification({
+    status: "error",
+    appearance: "negative",
+    content: message
+  });
+}
+
+// TODO: move inside of this function out
+function handleReleaseResponse(dispatch, json, release, revisions) {
+  if (json.success) {
+    // Update channel map based on the response
+    // We need to use channel_map_tree to get branches
+    Object.keys(json.channel_map_tree).forEach(trackKey => {
+      const track = json.channel_map_tree[trackKey];
+      Object.keys(track).forEach(seriesKey => {
+        const series = track[seriesKey];
+        Object.keys(series).forEach(archKey => {
+          const arch = series[archKey];
+          arch.forEach(map => {
+            if (map.revision) {
+              let revision;
+
+              if (map.revision === +release.id) {
+                // release.id is a string so turn it into a number for comparison
+                revision = release.revision;
+              } else if (revisions[map.revision]) {
+                revision = revisions[map.revision];
+              } else {
+                revision = {
+                  revision: map.revision,
+                  version: map.version,
+                  architectures: release.revision.architectures
+                };
+              }
+
+              let channel = map.channel;
+              if (RISKS.indexOf(channel.split("/")[0]) > -1) {
+                // TODO: this should be the default track, not "latest"
+                channel = `latest/${channel}`;
+              }
+
+              dispatch(releaseRevisionSuccess(revision, channel));
+            }
+          });
+        });
+      });
+    });
+  }
+}
+
+function fetchReleases(dispatch, csrfToken, snapName, releases, revisions) {
```

Comment:
I would say this should also be in api module, but not sure what to do with passing of `dispatch`.
Maybe instead of passing `dispatch` we could pass `successCallback` that would just be a function (`if (successCallback) successCallback(revision, channel)`). This way api module can be free of dependencies on dispatch, while still being able to call arbitrary callbacks on success/error?

Reply:
I guess we could return a list/object of the results from the queue and iterate over them within the action, rather than calling `handleReleaseResponse` within the `fetch` itself - give it one job?

Reply:
That's a good point. Maybe we could use `Promise.all` for that? 
Anyway, one benefit of doing it in here is that each response is handled when it returns and doesn't need to wait for others. Passing it all to array would mean we only handle all once we get all results back.